source/Stargate-Model/WebOrigin.class.st

```diff
@@ -0,0 +1,76 @@
+"
+I represent a web origin as defined in the RFC 6454.
+I have the format 
+	scheme '://' host [ ':' port ]
+with <scheme>, <host>, <port> from RFC 3986.
+	
+The // precedes the authority component of the uri, and thus I accept the schemes
+that may have the authority component. Currently #(http https) but this can be extended at will. 
+	
+References
+	https://tools.ietf.org/html/rfc6454
+	https://tools.ietf.org/html/rfc3986
+"
+Class {
+	#name : #WebOrigin,
+	#superclass : #Object,
+	#instVars : [
+		'scheme',
+		'host',
+		'port'
+	],
+	#category : #'Stargate-Model-CORS'
+}
+
+{ #category : #'instance creation' }
+WebOrigin class >> basedOn: aUrl [
+
+	( self hasValidOrigin: aUrl )
+		ifFalse: [ InstanceCreationFailed signal: ( '<1p> does not comply with a valid origin' expandMacrosWith: aUrl ) ].
```

Comment:
```suggestion
	AssertionChecker enforce: [ self hasValidOrigin: aUrl ]
		because: [ '<1p> does not comply with a valid origin' expandMacrosWith: aUrl ]
		raising: InstanceCreationFailed.
```